ml_tools/eolearn/ml_tools/train_test_split.py

```diff
@@ -64,28 +77,25 @@ def execute(self, eopatch, *, seed=None):
         :return: Input EOPatch with the train set mask.
         :rtype: EOPatch
         """
-        np.random.seed(seed)
+        if self.split_type in [TrainTestSplitType.PER_CLASS, TrainTestSplitType.PER_PIXEL]:
+            np.random.seed(seed)
 
         ftype, fname, new_name = self.feature
         data = eopatch[(ftype, fname)]
 
-        if self.split_type == 'per_class':
+        if self.split_type == TrainTestSplitType.PER_PIXEL:
+            rands = np.random.rand(*data.shape)
```

Comment:
as this task is currently designed to work with `MASK_TIMELESS` type, I would add to the `_parse_feature` method on line 58 the argument `allowed_feature_types={FeatureType.MASK_TIMELESS}`, to avoid mis-behaviours (e.g. if `MASK` is specified instead)

Reply:
That is not true. It should handle almost any feature type, probably with a few exceptions. In the unit tests there are cases for MASK_TIMELESS, LABEL_TIMELESS and DATA wich is tested using 4D arrays.